releases/2022-11-25-releasenotes.md

```diff
@@ -0,0 +1,154 @@
+---
+draft: false
+title: "Handle errors the GraphQL way and more results to your query"
+description: "Vendia Share updated the response status code and allowing up to 1000 query results"
+date: '2022-11-25'
+authors:
+  - Li Dai
+tags:
+- Improvements
+- Developer Experience
+- GraphQL
+---
+
+Vendia Engineers worked tirelessly to improve our product and ensure that we follow and set best practices for industrial standards. Here are the newest updates!
+
+# Error handling the way it should with [GraphQL](https://graphql.org/). 
+
+Errors are not scary as long as we handle them well. In the past Vendia's end point would return 4XX and 5XX status codes for any http request failures. But that has changed as we decided to follow GraphQL's [recommended approach](https://graphql.github.io/graphql-over-http/draft/#sec-Field-errors-encountered-during-execution) over http calls.
+
+In short, it means the end point will always return 2XX. But when error happens, error messages will be included in the response. 
+
+Here's two examples using `Curl`:
+
+1. A request with typo in it's query. `n` is intentionally removed in `listVendia_BlockItems`:
```

Comment:
Great call on including the example error scenarios!  In other release notes, we often show it using an image of the malformed query in GraphQL explorer and the output on the right.   That then also shows off our UI and makes it really clear to see the query and output in one image, side-by-side.  

It's probably good to only highlight that which changes for the customer, i.e. the 4xx errors that are now 2xx and the list of error codes that will display back for each type of error.

Reply:
This table has some example "bad queries" that you can use some of the examples - https://www.notion.so/vendia/Malformed-Graphql-Request-762307a4dbdd49c3a527a21301fcb763.

Going with this approach also then helps display the attributes that are included in the result set, e.g. the new set of error messages.

Reply:
For reference, here are the error codes: https://www.notion.so/vendia/GraphQL-API-Errors-fe83bc5b6f4240ae96dfcf7f3b664a91#fb23db622a8346ccb0ea8e3cf57b3d40

Reply:
table made